c/tree.c

```diff
@@ -1145,8 +1153,15 @@ LOCAL void encode_node(MMDBW_tree_s *tree, MMDBW_node_s *node,
 
     uint32_t left =
         htonl(record_value_as_number(tree, &(node->left_record), args));
+    if (0 == left) {
```

Comment:
Checking limits the issue to aliased nodes. It certainly eliminates more possible failures than checking that no node points to one node in particular (the root node).